### DISCUSSION TITLE
How to use Linux direct interface??

### DISCUSSION THREAD

**Discussion by kukougu** [waltligon/orangefs#94] [Q&A]
I install orangefs-2.9.7, orangefs-server-2.9.7 and orangefs-devel-2.9.7 in Centos 7.6 by EPEL repo. Then I start the orangefs server successfully and configure the /etc/pvfs2tab file. 
But when I run 
`LD_PRELOAD=/usr/lib64/libofs.so mkdir /a` and `LD_PRELOAD=/usr/lib64/libofs.so stat /a`
nothing happen and nothing printf.
I try the fs by the fuse it can work. So I just want to know how to use the linux direct interface? The interface will get the server and port infomation from /etc/pvfs2tab automatically?
Thanks a lot.

**Comment by dreynol** [comment]
Is there a particular reason you are wanting to use the Direct Interface? Do you just need basic filesystem access or are you looking to compile a specific program to use the Direct Interface?

Generally, for a basic installation we recommend starting with the Linux Kernel Module interface. It allows you to access the OrangeFS filesystem using all your standard Linux tools/commands just as you would normally use them, without having to worry about LD_PRELOAD and things like that. Instructions to help you get started can be found here: http://docs.orangefs.com/quickstart/quickstart-package/.

If you find that the kernel module doesn't provide you the performance you need or other requirements necessitate using the Direct Interface, we will probably have more questions about your setup but will try to help. You can also find instructions for the Direct Interface here: http://docs.orangefs.com/nix-clients/direct-interface/.

Also, if it is useful, a comparison of the various client interfaces can be found here: http://docs.orangefs.com/nix-clients/

Let us know if you run into any issues with the docs or have more questions.

– David

**Reply by kukougu** [reply]
Thanks a lot, I use Linux direct interface successfully by compiling the source code.
